modules/aws/ssm.go

```diff
@@ -68,3 +73,127 @@ func NewSsmClientE(t testing.TestingT, region string) (*ssm.SSM, error) {
 
 	return ssm.New(sess), nil
 }
+
+// WaitForSsmInstanceE waits until the instance get registered to the SSM inventory.
+func WaitForSsmInstanceE(t testing.TestingT, awsRegion, instanceID string, timeout time.Duration) error {
+	timeBetweenRetries := 2 * time.Second
+	maxRetries := int(timeout.Seconds() / timeBetweenRetries.Seconds())
+	description := fmt.Sprintf("Waiting for %s to appear in the SSM inventory", instanceID)
+
+	key := aws.String("AWS:InstanceInformation.InstanceId")
+	filterType := aws.String("Equal")
+	values := aws.StringSlice([]string{instanceID})
+	_, err := retry.DoWithRetryE(t, description, maxRetries, timeBetweenRetries, func() (string, error) {
+		client := NewSsmClient(t, awsRegion)
+		req, resp := client.GetInventoryRequest(&ssm.GetInventoryInput{
+			Filters: []*ssm.InventoryFilter{
+				{
+					Key:    key,
+					Type:   filterType,
+					Values: values,
+				},
+			},
+		})
+
+		if err := req.Send(); err != nil {
+			return "", err
+		}
+
+		if len(resp.Entities) != 1 {
+			return "", fmt.Errorf("%s is not in the SSM inventory", instanceID)
+		}
+
+		return "", nil
+	})
+
+	return err
+}
+
+// WaitForSsmInstance waits until the instance get registered to the SSM inventory.
+func WaitForSsmInstance(t testing.TestingT, awsRegion, instanceID string, timeout time.Duration) {
+	err := WaitForSsmInstanceE(t, awsRegion, instanceID, timeout)
+	require.NoError(t, err)
+}
+
+// CheckSsmCommand checks that you can run the given command on the given instance through AWS SSM. Returns stdout and stderr.
+func CheckSsmCommand(t testing.TestingT, awsRegion, instanceID, command string, timeout time.Duration) (string, string) {
+	stdout, stderr, err := CheckSsmCommandE(t, awsRegion, instanceID, command, timeout)
+	if err != nil {
+		t.Fatalf("failed to execute '%s' on %s (%v):]\n  stdout: %#v\n  stderr: %#v", command, instanceID, err, stdout, stderr)
+	}
+	return stdout, stderr
+}
+
+type result struct {
```

Comment:
Any reason not to make this a public `struct` with a name like `CommandOutput` and return that from the `CheckSsmCommand` methods (instead of two strings that force you to always look up what each one is)?